hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/staking/RewardCalculator.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl.staking;
+
+import static com.hedera.node.app.service.token.Units.HBARS_TO_TINYBARS;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.hapi.node.state.token.StakingNodeInfo;
+import com.hedera.node.app.service.mono.ledger.accounts.staking.StakePeriodManager;
+import com.hedera.node.app.service.token.ReadableStakingInfoStore;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class RewardCalculator {
+    private final StakePeriodManager stakePeriodManager;
+    private long rewardsPaid;
+
+    @Inject
+    public RewardCalculator(@NonNull final StakePeriodManager stakePeriodManager) {
+        this.stakePeriodManager = stakePeriodManager;
+    }
+
+    public void reset() {
+        rewardsPaid = 0;
+    }
+
+    public long computePendingReward(
+            @NonNull final Account account, @NonNull final ReadableStakingInfoStore stakingInfoStore) {
+        final var effectiveStart = stakePeriodManager.effectivePeriod(account.stakePeriodStart());
+        if (!stakePeriodManager.isRewardable(effectiveStart)) {
+            return 0;
+        }
+
+        final var addressBookId = calculateNodeAddressId(account.stakedNodeId());
+        final var stakingInfo = stakingInfoStore.get(addressBookId);
+        final var rewardOffered =
+                computeRewardFromDetails(account, stakingInfo, stakePeriodManager.currentStakePeriod(), effectiveStart);
+        return account.declineReward() ? 0 : rewardOffered;
+    }
+
+    public long rewardsPaidInThisTxn() {
```

Comment:
We don't need this .. I have it in`StakingAccountsCommitInterceptor`

Reply:
will remove in next PR, thanks!

Reply:
Will do in next PR 👍